### PR TITLE
Fix #5770: Add JavaDefined to tasty reflect

### LINKED ISSUE #5770: Missing JavaDefined flag in tasty reflect
The flag `JavaDefined` is missing in tasty reflect. 

Despite that `JavaDefined` is not defined in tasty format (it is not necessary), the flag is used in Dotty and Scala2 macros.

### PR DESCRIPTION
**Pull request by liufengyun**
Fix #5770: Add JavaDefined to tasty reflect

### CODE REVIEW

File: library/src/scala/tasty/reflect/SymbolOps.scala [hunk 1]
```diff
@@ -14,6 +14,9 @@ trait SymbolOps extends Core {
     /** Flags of this symbol */
     def flags(implicit ctx: Context): Flags
 
+    /** Does the symbol match the flags? */
+    def is(flags: Flags)(implicit ctx: Context): Boolean
```
Comment (nicolasstucki) [inline library/src/scala/tasty/reflect/SymbolOps.scala]: There is already an `is` on the flags. This is redundant and should be removed from here. If you really want it you can add this extension method in the utils.

Comment (liufengyun) [reply library/src/scala/tasty/reflect/SymbolOps.scala]: It seems that it makes more sense to remove `Flags.is` and keep it here, from the usability point of view.

Comment (liufengyun) [reply library/src/scala/tasty/reflect/SymbolOps.scala]: Import `util` in order to use `symbol.is(flags)` is not friendly for programmers.

Comment (liufengyun) [reply library/src/scala/tasty/reflect/SymbolOps.scala]: I think there is a better way to organize the tasty reflect internals such that macro authors can write `symbol.is(flags)` without an additional import. Currently, this seems impossible without redundancy in tasty reflect.

We addressed the same problem before while experimenting with macros. The idea is to separate compiler-facing interface and macro-author-facing interface. The former corresponds to `core`, they should be minimal & no duplicate, it's a contract with Scala compilers. The latter  corresponds to `util`, they aim for friendliness, it's a contract with macro authors. The separation also helps manage API evolution by segregating concerns.

Macro authors do not need to care about `core` (the abstract methods can be protected thus not accessible), the default interface for macro authors is `util`.

Comment (nicolasstucki) [reply library/src/scala/tasty/reflect/SymbolOps.scala]: Wouldn't this required duplication of the code of the `core` in the `util`?

Comment (liufengyun) [reply library/src/scala/tasty/reflect/SymbolOps.scala]: Just discussed with @nicolasstucki and he has some idea to do the refactoring, will address in a later PR.

**Comment by nicolasstucki** [comment]
You need to update the 2 `showFlags` methods in `scala.tasty.reflect.Printers`. Otherwise LGTM

**Pull request merged**